index.js

```diff
@@ -141,28 +194,53 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let lastserverd = null
+
+function takeATicketNumber(line) {
+  if (line.length == 0 && lastserverd == null) {
+    line.push(1);
+    lastserverd = 1;
+  } 
+  else {
+    line.push(lastserverd + 1);
+    lastserverd++;
+  }
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+  return `Welcome. You are ticket number ${lastserverd}`;
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+}
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 1.`
+
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log("%c----------", "color: red") 
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc(newline){
+  newLine.push(1);
+  return `Welcome. You are ticket number ${newLine[newLine.length] - 1 }`
+
+}
+
+console.log("Im not sure how to do the last one without googling 😢")
```

Comment:
I'm glad you're googling, I hope you came across some good old classic documentation :)! This is a tricky, bonus one.  Try watching Ian's scrimba video here: [https://scrimba.com/scrim/cLgEmZUZ](https://scrimba.com/scrim/cLgEmZUZ) and play around with his examples (literally, pause the video and start editing the editor shown in the video). Then come back and try to solve this